src/Agent.Worker/ExecutionContext.cs

```diff
@@ -478,20 +478,32 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
             Repositories = message.Resources.Repositories;
 
             // JobSettings
-            var checkouts = message.Steps?.Where(x => Pipelines.PipelineConstants.IsCheckoutTask(x)).ToList();
+            var checkouts = message.Steps?.Where(x => Pipelines.PipelineConstants.IsCheckoutTask(x)).Cast<Pipelines.TaskStep>().ToList();
             JobSettings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
             JobSettings[WellKnownJobSettings.HasMultipleCheckouts] = Boolean.FalseString;
             if (checkouts != null && checkouts.Count > 0)
             {
                 JobSettings[WellKnownJobSettings.HasMultipleCheckouts] = checkouts.Count > 1 ? Boolean.TrueString : Boolean.FalseString;
-                var firstCheckout = checkouts.First() as Pipelines.TaskStep;
+                var firstCheckout = checkouts.First();
                 if (firstCheckout != null && Repositories != null && firstCheckout.Inputs.TryGetValue(Pipelines.PipelineConstants.CheckoutTaskInputs.Repository, out string repoAlias))
                 {
                     JobSettings[WellKnownJobSettings.FirstRepositoryCheckedOut] = repoAlias;
-                    var repo = Repositories.Find(r => String.Equals(r.Alias, repoAlias, StringComparison.OrdinalIgnoreCase));
-                    if (repo != null)
+
+                    // Mark either the self repository as the primary repository if it was checked out,
+                    // or otherwise mark the first checked out repository as the primary repository.
+                    var selfCheckout = checkouts.FirstOrDefault(c => c.Inputs.TryGetValue(Pipelines.PipelineConstants.CheckoutTaskInputs.Repository, out string repoSelfAlias) &&
+                                                                        RepositoryUtil.IsPrimaryRepositoryName(repoSelfAlias));
+                    var firstCheckoutRepo = Repositories.Find(r => String.Equals(r.Alias, repoAlias, StringComparison.OrdinalIgnoreCase));
+                    var selfRepo = Repositories.Find(r => RepositoryUtil.IsPrimaryRepositoryName(r.Alias));
```

Comment:
Find method is used for Repository and the FirstOrDefault method for checkouts. Both methods work the same way. Wouldn't it be better to use one method for all lists for consistency?

Reply:
thank you, I have changed the Find calls to FirstOrDefault calls